kebechet/webhooks.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+# Kebechet
+# Copyright(C) 2018 Fridolin Pokorny
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""This is the webhook receiver..."""
+
+
+import os
+import logging
+import hmac
+import json
+
+import requests
+
+from flask import request, Blueprint, jsonify, current_app
+from git import Repo
+
+
+ENDPOINT_URL = os.getenv('KEBESCHET_MATTERMOST_ENDPOINT_URL', None)
+
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.DEBUG)
+
+webhook = Blueprint('webhook', __name__, url_prefix='')
+
+
+def notify_channel(message: str) -> None:
+    """Send message to Mattermost Channel."""
+    payload = {'text': message,
+               'icon_url': 'https://avatars1.githubusercontent.com/u/33906690'}
+
+    r = requests.post(ENDPOINT_URL, json=payload)
+    if r.status_code != 200:
+        _LOGGER.error(f"cant POST to {ENDPOINT_URL}")
+
+
+def handle_github_open_issue(issue: dict) -> None:
+    """Will handle with care."""
+    _LOGGER.info(f"An Issue has been opened: {issue['url']}")
+
+    if issue['title'].startswith('Automatic update of dependency'):
+        return
+
+    notify_channel(f"[{issue['user']['login']}]({issue['user']['url']}) just "
+                   f"opened an issue: [{issue['title']}]({issue['html_url']})... :glowstick:")
+
+
+def handle_github_open_pullrequest(pullrequest: dict) -> None:
+    """Will handle with care."""
+    _LOGGER.info(f"A Pull Request has been opened: {pullrequest['url']}")
+
+    if pullrequest['title'].startswith('Automatic update of dependency'):
+        return
+
+    notify_channel(f"[{pullrequest['user']['login']}]({pullrequest['user']['url']}) just "
```

Comment:
nitpicking.. `pullrequest['user']['html_url']` will be probably better here and other places :angel: